change-notes/1.21/analysis-csharp.md

```diff
@@ -4,11 +4,17 @@
 
 C# analysis now supports the extraction and analysis of many C# 8 features. For details see [Changes to code extraction](#changes-to-code-extraction) and [Changes to QL libraries](#changes-to-ql-libraries) below.
 
+## New queries
+
+| **Query**                                     | **Tags**                                             | **Purpose**                                                                                                                                                                 |
+|-----------------------------------------------|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Class defines a field that uses an ICryptoTransform class in a way that would be unsafe for concurrent threads (`cs/thread-unsafe-icryptotransform-field-in-class`) | concurrency, security, external/cwe/cwe-362 | Highlights classes with fields that make use of a static `System.Security.Cryptography.ICryptoTransform` object. Using these classes in concurrent threads is dangerous. It may result an error and may also give incorrect results. |
+| Potential usage of an object implementing ICryptoTransform class in a way that would be unsafe for concurrent threads (`cs/thread-unsafe-icryptotransform-captured-in-lambda`) | concurrency, security, external/cwe/cwe-362 | Highlights instances of classes where a field of type `System.Security.Cryptography.ICryptoTransform` is captured by a lambda, and appears to be used in a thread initialization method. |
+
```

Comment:
I hadn't noticed before now, but the query names are quite verbose. I would prefer something like

`Class defines a field that uses an ICryptoTransform class in a way that would be unsafe for concurrent threads` ->
`Thread-unsafe use of a static ICryptoTransform field`

and

`Potential usage of an object implementing ICryptoTransform class in a way that would be unsafe for concurrent threads` -> `Thread-unsafe capturing of an ICryptoTransform object`

If you agree, could we update the query names on this PR as well?

Reply:
As discussed separately, this change would be better made on `master`.